### DISCUSSION TITLE
Syntax error: value, object or array expected.

### DISCUSSION THREAD

**Discussion by jusfor** [open-source-parsers/jsoncpp#1456] [Q&A]
Hi, I've got some issue and I can't find a way to solve it. Could you help me? Yes, I know there is a issue with similar topic but this solutions hasn't work for me :( (<a href= https://github.com/open-source-parsers/jsoncpp/issues/1325>Mentioned issue #1325</a>)

<h2>My code:</h2>

```cpp
#include <iostream>
#include <json/json.h>
#include <fstream>

using namespace std;
int main() {
    Json::Value config;

    std::ifstream json_file("config.json", std::ifstream::binary);
    json_file >> config;

    cout << "Version: " << config["version"] << "\n";

    return 0;
}
```

<h2>JSON file:</h2>

```json
{
  "version": "1.0",
  "appName": "test",
  "ownerId": "22113434234"
}
```

<h2>Error:</h2>

```
 terminate called after throwing an instance of 'Json::RuntimeError'
   what():  * Line 1, Column 1
   Syntax error: value, object or array expected.
```

I tried other JSON files but still same issue. 
I will really be very grateful for fast help :pray:

**Comment by BillyDonahue** [comment]
Can you debug this? It looks to me like it should work.
Like can you add some instrumentation to print the `ifstream` contents and confirm that it's the data you expect it to contain?

**Comment by BetweenOneway** [comment]
Maybe It was because your file config.json is encoding with UTF-8 BOM. Try to transfer it to UTF-8. That's what I met.
